File: BaekJoon/solve/1/9996/9996.cpp
#include <iostream>

using namespace std;

int n;
string pattern, pre, suf;
string input;

int main(void) { 
  cin >> n;
  cin >> pattern;
  int pos = pattern.find('*');
  pre = pattern.substr(0, pos);
  suf = pattern.substr(pos + 1);
  
  for(int i = 0; i < n; i++) { 
    cin >> input;
    if(pre.size() + suf.size() > input.size()) { 
      cout << "NE" << '\n';
      continue;
    }

    if(input.substr(0, pre.size()) == pre && input.substr(input.size() - suf.size()) == suf) { 
      cout << "DA" << '\n';
    } else { 
      cout << "NE" << '\n';
    }
  }
  return 0;
}

File: BaekJoon/solve/1/9996/9996.swift

let n = Int(readLine()!)!
let pattern = readLine()!

let index = pattern.firstIndex(of: "*")!
let distance = pattern.distance(from: pattern.startIndex, to: index)

let pre = pattern.prefix(distance)
let suf = pattern.suffix(pattern.count - distance - 1)

for _ in 0..<n { 
  let input = readLine()!
  if input.count < pre.count + suf.count { 
    print("NE")
    continue
  }

  if input.prefix(pre.count) == pre && input.suffix(suf.count) == suf { 
    print("DA")
  } else { 
    print("NE")
  }
}